src/Microsoft.OData.Core/Build.NuGet/Microsoft.OData.Core.Nightly.Release.nuspec

```diff
@@ -22,7 +22,7 @@ OData .NET library is open source at http://github.com/OData/odata.net. Document
       <group targetFramework=".NETStandard2.0">
         <dependency id="Microsoft.Spatial" version="[$VersionFullSemantic$-Nightly$NightlyBuildVersion$]" />
         <dependency id="Microsoft.OData.Edm" version="[$VersionFullSemantic$-Nightly$NightlyBuildVersion$]" />
-        <dependency id="Microsoft.Bcl.AsyncInterfaces" version="5.0.0" />

```

Comment:
There are other dependencies deprecated. Any plan to update them together?

Reply:
@xuzhg I upgrade other deprecated dependencies.

However, I couldn't upgrade **Microsoft.NETCore.App** deprecated dependency on different projects since it's a target framework dependency. Upgrading would mean dropping support for the framework. The deprecated dependency will be dropped in the next ODL major release.

- I replaced the deprecated **Microsoft.CodeAnalysis.FxCopAnalyzers** dependency with **Microsoft.CodeAnalysis.NetAnalyzers 7.0.4**

- I upgraded the deprecated **Microsoft.Extensions.DependencyInjection 2.2.0** on **Microsoft.Test.OData.DependencyInjection** project for `net472`, `netcoreapp2.1` and `netcoreapp3.1` target frameworks to **3.1.32** (supported) and not the latest **7.0.0** due to the following warnings observed in the case of 7.0.0:
  - _Microsoft.Extensions.DependencyInjection.Abstractions.targets(4,5): warning : Microsoft.Extensions.DependencyInjection.Abstractions 7.0.0 doesn't support netcoreapp2.1 and has not been tested with it. Consider upgrading your TargetFramework to net6.0 or later. You may also set <SuppressTfmSupportBuildWarnings>true</SuppressTfmSupportBuildWarnings> in the project file to ignore this warning and attempt to run in this unsupported configuration at your own risk._
  - _Microsoft.Extensions.DependencyInjection.targets(4,5): warning : Microsoft.Extensions.DependencyInjection 7.0.0 doesn't support netcoreapp2.1 and has not been tested with it. Consider upgrading your TargetFramework to net6.0 or later. You may also set <SuppressTfmSupportBuildWarnings>true</SuppressTfmSupportBuildWarnings> in the project file to ignore this warning and attempt to run in this unsupported configuration at your own risk._
  - _Microsoft.Extensions.DependencyInjection.Abstractions.targets(4,5): warning : Microsoft.Extensions.DependencyInjection.Abstractions 7.0.0 doesn't support netcoreapp3.1 and has not been tested with it. Consider upgrading your TargetFramework to net6.0 or later. You may also set <SuppressTfmSupportBuildWarnings>true</SuppressTfmSupportBuildWarnings> in the project file to ignore this warning and attempt to run in this unsupported configuration at your own risk._
  - _Microsoft.Extensions.DependencyInjection.targets(4,5): warning : Microsoft.Extensions.DependencyInjection 7.0.0 doesn't support netcoreapp3.1 and has not been tested with it. Consider upgrading your TargetFramework to net6.0 or later. You may also set <SuppressTfmSupportBuildWarnings>true</SuppressTfmSupportBuildWarnings> in the project file to ignore this warning and attempt to run in this unsupported configuration at your own risk._

- I couldn't upgrade **Microsoft.Extensions.DependencyInjection** dependency on **Microsoft.Test.OData.DependencyInjection** project for `net45` and `netcoreapp1.1` target frameworks from **1.1.0** to **3.1.32** or higher since the two frameworks only support 1.x versions of the library. The deprecated dependency will be dropped in the next ODL major release.